standalone/src/models/product-selection/builders.spec.ts

```diff
@@ -0,0 +1,144 @@
+import { TProductSelectionGraphql, TProductSelectionRest } from './types';
+import {
+  ProductSelection,
+  ProductSelectionGraphql,
+  ProductSelectionRest,
+} from './index';
+
+const validateRestModel = (model: TProductSelectionRest): void => {
+  expect(model).toMatchObject({
+    version: expect.any(Number),
+    id: expect.any(String),
+    createdAt: expect.any(String),
+    lastModifiedAt: expect.any(String),
+    lastModifiedBy: expect.objectContaining({
+      externalUserId: expect.any(String),
+      anonymousId: expect.any(String),
+      clientId: expect.any(String),
+      customer: expect.objectContaining({
+        id: expect.any(String),
+        typeId: 'customer',
+        obj: expect.objectContaining({
+          id: expect.any(String),
+        }),
+      }),
+    }),
+    createdBy: expect.objectContaining({
+      externalUserId: expect.any(String),
+      anonymousId: expect.any(String),
+      clientId: expect.any(String),
+      customer: expect.objectContaining({
+        id: expect.any(String),
+        typeId: 'customer',
+        obj: expect.objectContaining({
+          id: expect.any(String),
+        }),
+      }),
+    }),
+    key: expect.any(String),
+    productCount: expect.any(Number),
+    mode: expect.any(String),
+    custom: expect.any(Object),
+    name: expect.objectContaining({
+      en: expect.any(String),
+    }),
+  });
+};
+
+const validateGraphqlModel = (model: TProductSelectionGraphql): void => {
+  expect(model).toMatchObject({
+    version: expect.any(Number),
+    id: expect.any(String),
+    createdAt: expect.any(String),
+    lastModifiedAt: expect.any(String),
+    lastModifiedBy: expect.objectContaining({
+      __typename: 'Initiator',
+      externalUserId: expect.any(String),
+      anonymousId: expect.any(String),
+      clientId: expect.any(String),
```

Comment:
There's no need to validate specific functions of nested models in a GraphQL one. Asserting the `__typename` property should be enough.

Reply:
Makes sense, we already tested those fields in the submodel's test 👍🏻